node/src/p2p/callbacks/p2p_callbacks_reducer.rs

```diff
@@ -290,6 +296,43 @@ impl crate::State {
                     best_tip: best_tip.clone(),
                 });
             }
+            P2pCallbacksAction::P2pPubsubValidateMessage {
+                message,
+                message_content,
+                peer_id,
+            } => {
+                let result = match message_content {
+                    GossipNetMessageV2::NewState(new_best_tip) => {
+                        match BlockWithHash::try_new(new_best_tip.clone()) {
+                            Ok(block) => {
+                                let allow_block_too_late = allow_block_too_late(state, &block);
+                                match state.prevalidate_block(&block, allow_block_too_late) {
+                                    Ok(()) => ValidationResult::Valid,
```

Comment:
At this point we only know that the block is well-formed, but that doesn't mean that it is valid. We can only consider it valid *after* it has been successfully applied.